File: python/leetcode/P12.py
class Solution:
	def intToRoman(self, num: int) -> str:
		roman_dict = {1000: 'M', 900: 'CM', 500: 'D', 400: 'CD', 100: 'C', 90: 'XC', 50: 'L', 40: 'XL', 10: 'X', 9: 'IX', 5: 'V', 4: 'IV', 1: 'I'}
		roman = ''
		for n in roman_dict.keys():
			roman += roman_dict[n] * (num // n)
			num %= n
		return roman

File: java/leetcode/P12.java
import java.util.HashMap;
import java.util.LinkedHashMap;

public class P12 {
	public String intToRoman(int num) {
		HashMap<Integer, String> romanTable = new LinkedHashMap<>() {{ put(1000, "M"); put(900, "CM"); put(500, "D"); put(400, "CD"); put(100, "C"); put(90, "XC"); put(50, "L"); put(40, "XL"); put(10, "X"); put(9, "IX"); put(5, "V"); put(4, "IV"); put(1, "I"); }};
		String roman = "";
		for (Integer i : romanTable.keySet()) {
			int count = num / i;
			num %= i;
			for (int j = 0; j < count; j++) {
				roman += romanTable.get(i);
			}
		}
		return roman;
	}
}
